pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/util/StringSuggestionTypeaheadComparator.java

```diff
@@ -0,0 +1,72 @@
+package org.broadinstitute.ddp.util;
+
+import java.util.Comparator;
+import java.util.regex.Pattern;
+
+/**
+ * Comparator that can be used to sort results for
+ * the picklist option typeahead.  The general approach is favoring leftwards
+ * occurrences, in particular word starts, allowing for a starting
+ * parentheses.
+ */
+public class StringSuggestionTypeaheadComparator implements Comparator<String> {
+
+    private final String searchTerm;
+
+    /**
+     * Create a new one, using the given search term
+     * @param searchTerm a simple search term, provided by the
+     *                   user.  Not a regex.
+     */
+    public StringSuggestionTypeaheadComparator(String searchTerm) {
+        this.searchTerm = searchTerm.toUpperCase();
+    }
+
+    @Override
+    public int compare(String string1, String string2) {
```

Comment:
search logic to place matches with start of work first
pretty much logic used in AUTOCOMPLETE